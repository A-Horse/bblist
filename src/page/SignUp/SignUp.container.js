import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import Actions, { makeActionRequestCollection } from '../../actions/actions';
import SignUp from './SignUp';

const mapDispatchToProps = dispatch => {
  return {
    actions: bindActionCreators(
      makeActionRequestCollection([Actions.SIGNUP]),
      dispatch
    )
  };
};

const mapStateToProps = state => {
  return {
    signUpSuccess: state.auth.get('signUpSuccess')
  };
};

const SignUpContainer = withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(SignUp)
);

export default SignUpContainer;
