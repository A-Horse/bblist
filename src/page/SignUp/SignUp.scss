@import "../../style/variable";
@import "../../style/mixin";

// TODO refactor with signin
.signup-page {
  position: absolute;
  left: 0;
  top: 0;
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: center;

  .signup-main {
    width: 330px;
    align-self: center;
  }

  .signin-tip {
    margin-top: 18px;
    display: flex;
    justify-content: space-between;
    color: $black-light;

    .signin-link {
      font-weight: bolder;
    }
  }

  .signup-form {
    margin-top: 18px;
    width: 330px;
    align-self: center;

    input {
      @include form-input;
    }

    .error-msg {
      margin-bottom: 25px;
    }

    .signup-button {
      width: 100%;
    }
  }
}
