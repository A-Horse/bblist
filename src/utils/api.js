import { API_PREFIX, LOCAL_BACKUP_URL } from '../constants';
import { Storage } from 'services/storage';

export function makeApiUrl(path) {
  if (Storage.get(LOCAL_BACKUP_URL)) {
    return `${Storage.get(LOCAL_BACKUP_URL)}${path}`;
  }
  return `${API_PREFIX}${path}`;
}
