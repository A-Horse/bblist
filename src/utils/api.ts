export function makeApiUrl(path) {
  return `/api${path}`;
}
