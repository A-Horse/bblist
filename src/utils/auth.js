//
import { JWT_STORAGE_KEY, CACHED_USERNAME, CACHED_USERID, CACHED_USEREMAIL } from '../constants';
import { Storage } from '../services/storage';

export function getJWT() {
  return Storage.get(JWT_STORAGE_KEY);
}

export function saveJWT(jwt) {
  return Storage.set(JWT_STORAGE_KEY, jwt);
}

export function getCachedUserId() {
  return Storage.get(CACHED_USERID);
}

function saveUserData(userData) {
  return Storage.set('USER_DATA', userData);
}

export function getUserData() {
  const userData = Storage.get('USER_DATA');
  return userData ? JSON.parse(userData) : null;
}


export function saveAuthData(response) {
  Storage.set(CACHED_USERID, response.user.id);
  Storage.set(CACHED_USERNAME, response.user.username);
  Storage.set(CACHED_USEREMAIL, response.user.email);

  saveUserData(JSON.stringify(response.user));

  saveJWT(response.token);
}
