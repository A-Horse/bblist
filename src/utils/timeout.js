import { Promise } from "es6-promise";

export function timeout(ms) {
  return new Promise(resolve => setTimeout(resolve, ms));
}
