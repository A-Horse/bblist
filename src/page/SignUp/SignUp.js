import React, { Component } from 'react';
import { validateFormValue } from '../../services/validate-strategy';
import { PageContainer } from '../../components/widget/PageContainer';
import { Input } from '../../components/widget/Input/Input';
import { updateTitle } from '../../services/title';
import { Button } from '../../components/widget/Button/Button';
import { LogoBan } from '../../components/commons/LogoBan/LogoBan';
import { ErrorMsg } from '../../components/ErrorMsg/ErrorMsg';
import { Link } from 'react-router-dom';
import * as R from 'ramda';

import './SignUp.scss';

class SignUp extends Component {
  state = {
    errorMessage: {}
  };

  constructor(props) {
    super(props);
    this.signup = this.signup.bind(this);
  }

  componentWillMount() {
    updateTitle('Sign Up');
  }

  componentWillReceiveProps(newProps) {
    if (newProps.signUpSuccess) {
      this.props.actions.SIGNUP_FINISH();
      this.props.history.push('/signin');
    }
  }

  render() {
    const errorMessages = this.state.errorMessages;

    return (
      <PageContainer className="signup-page">
        <div className="signup-main">
          <LogoBan />
          <form className="signup-form" onSubmit={this.signup}>
            <div>
              <Input
                type="text"
                ref={ref => (this.emailInput = ref)}
                name="octopus-email"
                required
                placeholder="Email"
              />
            </div>

            <div>
              <Input
                type="text"
                ref={ref => (this.userNameInput = ref)}
                name="octopus-username"
                required
                placeholder="Name"
              />
            </div>

            <div>
              <Input
                type="password"
                ref={ref => (this.passwordInput = ref)}
                name="octopus-password"
                required
                placeholder="Password"
              />
            </div>

            <div>
              <Input
                type="password"
                ref={ref => (this.confirmPasswordInput = ref)}
                name="octopus-confirm-password"
                required
                placeholder="Password ConfirmPassword"
              />
            </div>

            <ErrorMsg messages={R.values(errorMessages)} />

            <Button className="signup-button" type="submit" styleType="primary" size="large">
              Sign Up
            </Button>
          </form>

          <div className="signin-tip">
            Already has an Account?
            <Link className="signin-link" to="/signin">
              Sign In
            </Link>
          </div>
        </div>
      </PageContainer>
    );
  }

  signup(event) {
    event.preventDefault();

    const signUpData = {
      name: this.userNameInput.value.trim(),
      password: this.passwordInput.value.trim(),
      email: this.emailInput.value.trim(),
      confirmPassword: this.confirmPasswordInput.value.trim()
    };

    const errorMessages = validateFormValue(signUpData, {
      name: [
        'max@100#Name Up to 100 characters ',
        'min@3#The name must be a minimum of three characters'
      ],
      password: ['max@100#Password Up to 100 characters', 'min@6#password min 6'],
      confirmPassword: [`eqTo@${signUpData.password}#password don't match`],
      email: ['email#email express wrong', 'max@150#max 150']
    });

    this.setState({ errorMessages: errorMessages });

    if (Object.keys(errorMessages).length) {
      return;
    }
    this.props.actions.SIGNUP_REQUEST(signUpData);
  }
}

export default SignUp;
