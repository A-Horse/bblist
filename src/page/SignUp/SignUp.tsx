import './SignUp.scss';
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { LogoBan } from '../../components/LogoBan/LogoBan';
import { Input } from '../../widget/Input/Input';
import { updateTitle } from '../../services/title';
import { AppButton } from '../../widget/Button';

class SignUp extends Component<any, any> {
  state = {
    errorMessage: {},
    username: '',
    password: '',
    passwordConfirm: '',
    email: '',
  };

  constructor(props: any) {
    super(props);
    this.signup = this.signup.bind(this);
  }

  componentWillMount(): void {
    updateTitle('Sign Up');
  }

  componentWillReceiveProps(newProps: any) {
    if (newProps.signUpSuccess) {
      this.props.actions.SIGNUP_FINISH();
      this.props.history.push('/login');
    }
  }

  render() {
    return (
      <div className="signup-main">
        <LogoBan />
        <form className="signup-form" onSubmit={this.signup}>
          <div>
            <Input
              value={this.state.email}
              onChange={(value) => this.setState({ email: value })}
              type="text"
              name="email"
              required
              placeholder="Email"
            />
          </div>

          <div>
            <Input
              value={this.state.username}
              onChange={(value) => this.setState({ username: value })}
              type="text"
              name="username"
              required
              placeholder="Name"
            />
          </div>

          <div>
            <Input
              value={this.state.password}
              onChange={(value) => this.setState({ password: value })}
              type="password"
              name="password"
              required
              placeholder="Password"
            />
          </div>

          <div>
            <Input
              value={this.state.passwordConfirm}
              onChange={(value) => this.setState({ passwordConfirm: value })}
              type="password"
              name="password-confirm"
              required
              placeholder="Confirm Password"
            />
          </div>

          <AppButton className="signup-button" htmlType="submit">
            Sign Up
          </AppButton>
        </form>

        <div className="signin-tip">
          Already has an Account?
          <Link className="signin-link" to="/signin">
            Sign In
          </Link>
        </div>
      </div>
    );
  }

  signup(event: any) {
    event.preventDefault();

    const signUpData = {
      username: this.state.username,
      email: this.state.email,
      password: this.state.password,
    };

    this.props.actions.SIGNUP_REQUEST(signUpData);
  }
}

export default SignUp;
