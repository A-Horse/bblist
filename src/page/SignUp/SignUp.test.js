import React from "react";
import SignUp from "./SignUp";
import { timeout } from "utils/timeout";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";
import { fromJS } from "immutable";
import { mount } from "enzyme";

test("SignUp", () => {
  const actions = {};
  const history = {};

  const tree = renderer.create(
    <MemoryRouter>
      <SignUp actions={actions} history={history} />
    </MemoryRouter>
  );
  expect(tree).toMatchSnapshot();
});

test("SignUp signup", () => {
  const actions = { SIGNUP_REQUEST: jest.fn() };
  const history = { push: jest.fn() };

  const wrapper = mount(
    <MemoryRouter>
      <SignUp actions={actions} history={history} />
    </MemoryRouter>
  );

  const emailInput = wrapper.find('input[name="octopus-email"]');
  const userNameInput = wrapper.find('input[name="octopus-username"]');
  const passwdInput = wrapper.find('input[name="octopus-password"]');
  const cotopusConfirmPasswordInput = wrapper.find(
    'input[name="octopus-confirm-password"]'
  );

  emailInput.simulate("focus");
  emailInput.node.value = "octopus@octopus.com";
  emailInput.simulate("change");

  userNameInput.simulate("focus");
  userNameInput.node.value = "octos";
  userNameInput.simulate("change");

  passwdInput.simulate("focus");
  passwdInput.node.value = "octopus";
  passwdInput.simulate("change");

  cotopusConfirmPasswordInput.simulate("focus");
  cotopusConfirmPasswordInput.node.value = "octopus";
  cotopusConfirmPasswordInput.simulate("change");

  wrapper
    .find(".signup-form")
    .simulate("focus")
    .simulate("submit");

  expect(actions.SIGNUP_REQUEST.mock.calls.length).toBe(1);
  expect(actions.SIGNUP_REQUEST).toBeCalledWith({
    name: "octos",
    email: "octopus@octopus.com",
    password: "octopus",
    confirmPassword: "octopus"
  });
});
