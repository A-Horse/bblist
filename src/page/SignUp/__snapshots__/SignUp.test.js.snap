// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp 1`] = `
<div
  className="page-container signup-page"
>
  <div
    className="signup-main"
  >
    <div
      className="logoban"
      onClick={[Function]}
    >
      <img
        className="logo"
        src="/static/logo.png"
      />
      <img
        className="text-logo"
        src="/static/text-logo.svg"
      />
    </div>
    <form
      className="signup-form"
      onSubmit={[Function]}
    >
      <div>
        <input
          name="octopus-email"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="text"
        />
      </div>
      <div>
        <input
          name="octopus-username"
          onChange={[Function]}
          placeholder="Name"
          required={true}
          type="text"
        />
      </div>
      <div>
        <input
          name="octopus-password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
        />
      </div>
      <div>
        <input
          name="octopus-confirm-password"
          onChange={[Function]}
          placeholder="Password ConfirmPassword"
          required={true}
          type="password"
        />
      </div>
      <noscript />
      <button
        className="button large primary border-default  signup-button"
        disabled={undefined}
        onClick={undefined}
        type="submit"
      >
        Sign Up
      </button>
    </form>
    <div
      className="signin-tip"
    >
      Already has an Account?
      <a
        className="signin-link"
        href="/signin"
        onClick={[Function]}
      >
        Sign In
      </a>
    </div>
  </div>
</div>
`;
